### ISSUE TITLE
[java] LawOfDemeter: False positive for standard UTF-8 charset name

### ISSUE THREAD

**Issue by mcandre** [pmd/pmd#1605]
**Affects PMD Version:** 

6

**Rule:**

LawOfDemeter

**Description:**

PMD's application of the Law of Demeter considers it a violation to query the standard name of the Java UTF-8 character set. Unless the character set is assigned to an explicit variable name first.

**Code Sample demonstrating the issue:**

```
public final class Fib {
    /** utility class */
    private Fib() {}

    public static void main(final String[] args) throws IOException {
        System.out.print("N = ");

        try (Scanner scanner = new Scanner(System.in, StandardCharsets.UTF_8.name())) {
            final int n = scanner.nextInt();
            System.out.println(String.format("Fib %d = %d", n, fib(n)));
        }
    }
}
```

As a workaround, I am disabling this rule in my rulesets.xml:

```xml
<rule ref="category/java/design.xml">
    <exclude name="LawOfDemeter" />
</rule>
```

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

Gradle

**Comment by Fernal73** [comment]
You could create an instance variable that points to StandardCharsets.UTF_8.name() and that would resolve this false positive.

**Comment by jsotuyod** [comment]
This is fixed in PMD 7.0.0-rc1. It won't be backported to PMD 6.
